pkg/controller/configmap/config.go

```diff
@@ -266,26 +270,28 @@ func HasNetworkConfigChange(currConfig *configv1.Network, prevConfig *configv1.N
 	return stringSliceEqual(currCidrs, prevCidrs)
 }
 
-func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap) (ncpNeedChange bool, agentNeedChange bool, err error) {
+func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap) (
+	ncpNeedChange bool, agentNeedChange bool, bootstrapNeedChange bool, err error) {
```

Comment:
do you think there might be a case for creating a struct describing which components need a configuration change?

Reply:
I added a map for bootstrap pod in pkg/types/defaults.go, do you mean to add a whole list for all components?

Reply:
I did not mean the section.
I notice that the function is now return a tuple of 4 values, and I was wondering it might be more manageable to have the 3 boolean in a struct

Reply:
OK, sure